### ISSUE TITLE
Timestamp of corrupted events get shown as First Timestamp

### ISSUE THREAD

**Issue by YamatoSecurity** [Yamato-Security/hayabusa#1095]
We talked by DM about this.
When a record's timestamp is corrupted, the First Timestamp will be incorrect. In the log that I sent you, in the past it would say `First Timestamp: +67137-05-07` but now it gets displayed as `1601-05-07` on my system. (I am not sure what changed..)
I checked the results, and Hayabusa does not output any events with a `1601-05-07` date so I think we can just ignore this timestamp when calculating the `First Timestamp`.
(I am not sure how we can do that though.)

**Comment by fukusuket** [comment]
@YamatoSecurity 
I have investigated broken evtx!
Certainly, this evtx was data that could not be opened with `Event Viewer`, `PowerShell`, or the `wevtutil` command...

When I tried to output all records with the `search` command, there was actually `1601-05-06` timestamp record 😲

```
fukusuke@fukusukenoAir hayabusa-2.6.0-all-platforms % ./hayabusa-2.6.0-mac-arm search --regex ".*"  -f ../Application.evtx -o out.csv -C -q

...

Elapsed time: 00:00:01.101
Errors were generated. Please check ./logs/errorlog-20230617_130121.log for details.

fukusuke@fukusukenoAir hayabusa-2.6.0-all-platforms % cat out.csv | grep 1601-
"1601-05-06T23:42:21.278515Z","DCX1.atqinsurance.local","App",25856,"-","-","-","../Application.evtx"
```

I think Hayabusa's current behavior is correct because this evtx record  is correct as a date format.
It's an unnatural date... :( , but I think it's better to output the date data in evtx as it is.

For the above reason, I think it is better not to change the current behavior. What do you think?🤔

**Comment by YamatoSecurity** [comment]
@fukusuket Thanks for investigating this! Although indeed it is a date existing in the log, it is obviously corrupt and invalid and there is no useful data in it anyways so I want to ignore these events. It is better for the analyst to know when the real start of the events happened. I am not sure what the most efficient way to filter on this is though. `.evtx` was released in Windows vista on January 30, 2007 so there shouldn't be a date before this. One way might be to check if the date is 2006 or earlier and print a warning and maybe leave a log in the error log if there is a date before then.

**Comment by fukusuket** [comment]
@YamatoSecurity 
I've got it! So, I'm thinking of implementing it with the following specifications, is this okay?
- Ignore corrupt records when calculating the `First Timestamp` if the date is older than January 30, 2007
- Output to error log if the date is older than January 30, 2007

**Comment by YamatoSecurity** [comment]
@fukusuket 
Thanks!
I think that is a good way to deal with this issue.

**Comment by hitenkoku** [comment]
closed via #1102
